services/consensusalgo/leanhelix/algorithm.go

```diff
@@ -22,33 +21,34 @@ func (s *service) leaderProposeNextBlockIfNeeded() error {
 		return nil
 	}
 
-	proposedTransactions, err := s.transactionPool.GetTransactionsForOrdering(&services.GetTransactionsForOrderingInput{
-		MaxNumberOfTransactions: 1,
+	txOutput, err := s.consensusContext.RequestNewTransactionsBlock(&services.RequestNewTransactionsBlockInput{
+		BlockHeight:             s.lastCommittedBlockHeight + 1,
+		MaxBlockSizeKb:          0,
+		MaxNumberOfTransactions: 0,
+		PrevBlockHash:           nil,
 	})
 	if err != nil {
 		return err
 	}
 
+	txBlock := txOutput.TransactionsBlock
+	txBlock.BlockProof = (&protocol.TransactionsBlockProofBuilder{}).Build()
+
+	rxOutput, err := s.consensusContext.RequestNewResultsBlock(&services.RequestNewResultsBlockInput{
+		BlockHeight:       s.lastCommittedBlockHeight + 1,
+		PrevBlockHash:     nil,
```

Comment:
It's also a smell left on purpose so we remember to fill in this field. Right now we just refactored the minimum to keep the acceptance working in lean helix and this wasn't required to make it pass.

The reason it's not needed is that lean helix today doesn't verify the prev block hash because it currently has a very partial implementation